web/ui/static/js/graph/index.js

```diff
@@ -1143,6 +1145,26 @@ function escapeHTML(string) {
   });
 }
 
+function limitMetrics(result) {
+  var MAX_METRICS_NUM = 10000;
+  var message = "<strong>Warning!</strong> Fetched " +
+    result.length + " metrics, but displaying only first " +
+    MAX_METRICS_NUM + " ones.";
+
+  if ($("#max-metrics-warning")[0]) {
+    // If message already exists
+    $("#max-metrics-warning").remove();
```

Comment:
From a UX perspective, I think this should be done when you submit a query.

Reply:
Done

Reply:
Please don't merge it yet. I'm refactoring it a bit.

Reply:
Done